src/p2p.rs

```diff
@@ -124,10 +128,16 @@ impl Connection {
                     );
                     Ok(block)
                 })?;
-                self.blocks_duration
-                    .observe_duration("process", || func(hash, block));
+                result.push((hash, block));
```

Comment:
I'm afraid it may result in OOM when there are many blocks to scan when subscribing to a "popular" address.

Reply:
Yeah, that is a legitimate worry...
I tried to spawn as soon as the first block arrives, which would also be a performance improvement like I said in my first comment. But I have a compile issue I can't solve https://github.com/romanz/electrs/commit/9443888fd084f5562857d1ccdb3b5fddb8418faa . Rayon docs https://docs.rs/rayon/latest/rayon/fn.scope.html even has a specific section about it, but the only suggestion avoiding the move is not working.
I may try again or use a less elegant solution of processing by chunks of blocks (like 16 or so) with the par iter

Reply:
@romanz should be fixed in the last commit (thanks for the suggestion @afilini)